### DISCUSSION TITLE
Run ezBEQ on VM as a Service

### DISCUSSION THREAD

**Discussion by d45p00k13** [3ll3d00d/ezbeq#69] [Q&A] [answered]
Dear Team,

I already installed a virtual machine in order to run ezBEQ on it. It runs flawlessly.

As I want to avoid the need to log into it to run ezBEQ in user context, i tried to prepare a task through windows 11 task scheduler to run it in user context when the system boots (due to time schedule after suspend/shutdown the NAS most likely) but obviously this does not work right away.

Is there a way to debug the problem or is it known at all; should it work and if so, is there something I might missed?

Thanks for your help,
D45p00k13

**Comment by 3ll3d00d** [comment]
It should be logging to $home/.ezbeq/ezbeq.log so you can look there, if nothing there then it's failing before that point somewhere in windows. No known problem with doing this, I have run such apps on startup in a similar fashion before without a problem.

**Comment by d45p00k13** [comment, marked as answer]
Dear 3ll3d00d,

thanks for your response! Again! As you helped me with running the BEQ designer in the VM in the first place ;)

As you expected, the problem was the configuration of the task upfront. Using the task scheduler in the first place was nothing I found very comfortable so I looked for an alternative. There are some but I found most easy-to-use the non-sucking service manager (NSSM) as a wrapper for a windows service to be able to monitor, stop and start the service from the windows task manager.

As I use an VM on a NAS, my executables were on network shares, but this brought some difficulties (like UNC paths are not allowed). So to make things clear, I will write down what to do, even if it looks some kind of trivial to some:

(Step 0. Make sure you installed BEQ Designer and if problems occur the latest vc redist package)
Step 1. Make a directory like C:\BEQ
Step 2. Copy ezBEQ, minidsp-rs, and the NSSM executables into the directory
Step 3. Start ezbeq to create the corresponding folder $home/.ezbeq/ and the configuration
Step 4. Edit the .yml file accordingly
 - the correct folder from Step 1 for the exe
 - options: 'dongles-ip-address:5333' --- if you use the WI-DG dongle from miniDSP
 - device_type: DDRC24 --- add it in case you use a miniDSP DDRC-24. For other models you must refer to the manual.
 - edit host and iconPath

Step 5. Test local installation --- if you run ezbeq.exe you should be able to access and use 127.0.0.1:8080. Try it!
Step 6. Go to the terminal console, entering C:\BEQ
Step 7. Enter: nssm.exe install ezbeq
- GUI opens up, add a description if you want
- select path and ezbeq.exe
- make sure the service runs in the user context and not using local system account (so enter username and password)

Congratulations, you should have the ezBEQ up and running on system | VM startup.
